test/integration/targets/jinja2_native_types/test_casting.yml

```diff
@@ -0,0 +1,24 @@
+- name: cast things to other things
+  set_fact:
+      int_to_str: "{{ i_two|to_text }}"
+      str_to_int: "{{ s_two|int }}"
+      dict_to_str: "{{ dict_one|to_text }}"
+      list_to_str: "{{ list_one|to_text }}"
+      int_to_bool: "{{ i_one|bool }}"
+      str_true_to_bool: "{{ s_true|bool }}"
+      str_false_to_bool: "{{ s_false|bool }}"
+
+- assert:
+    that:
+        - 'int_to_str == "2"'
+        - 'int_to_str|type_debug in ["string", "unicode"]'
+        - 'str_to_int == 2'
+        - 'str_to_int|type_debug == "int"'
+        - 'dict_to_str|type_debug in ["string", "unicode"]'
+        - 'list_to_str|type_debug in ["string", "unicode"]'
```

Comment:
So I am proposing to Jinja a set of data type tests to make type-related expressions easier and more strict. https://github.com/pallets/jinja/pull/824

I see that this could probably take advantage of it as well.

```yaml
    - int_to_str is string
    - str_to_int is integer
    - list_to_str is string
    - int_to_bool is boolean
    - str_true_to_bool is true
```
because currently doing something like:
```yaml
    - int_to_bool == True
```
would work fine for values 1 or 1.0

It is probably safer to write today:
```yaml
    - int_to_bool is sameas true
    - str_true_to_bool is sameas true
```
See https://github.com/dagwieers/ansible/blob/jinja2-type-tests/test/integration/targets/jinja2_tests/tasks/main.yml